### PR TITLE
release-notes: add release notes for the v95 release

### PR DESCRIPTION
**Pull request by angolini**
RN: Add the initial version for the v95 Release Notes
missing: 
- Link for the test results
- [FAQ](https://docs.foundries.io/latest/user-guide/troubleshooting/troubleshooting.html#jailhousesupportin-lmp-is-removed)
- [FAQ](https://docs.foundries.io/latest/user-guide/troubleshooting/troubleshooting.html#update-foundriesfactory-fanifest-merge-conflict)

### CODE REVIEW

**Comment by angolini** [comment]
See here the "rendered" markdown https://github.com/foundriesio/docs/blob/rn95/release-notes/rn_v95.md

**Comment by detsch** [comment]
> Two different ways to display the same information:
> 
> [1] https://github.com/foundriesio/docs/blob/5e8518539e119bd899da13163af5d8a0b7226636/release-notes/rn_v95.md
> 
> [2] https://github.com/foundriesio/docs/blob/e6c768424ccacc923654eb27cc0622d8117ef4c2/release-notes/rn_v95.md
> 
> Which one do you prefer?

My vote goes for [2]

File: release-notes/rn_v95.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    - [Improvements](#improvements)
+    - [Bug Fixes](#bug-fixes)
+    - [Testing](#testing)
+  - [Composectl Updates](#composectl-updates)
+    - [New Features](#new-features-1)
+    - [Bug Fixes](#bug-fixes-1)
+    - [Testing](#testing-1)
+  - [General Updates](#general-updates)
+    - [Deprecation list](#deprecation-list)
+  - [Plans for the FUture](#plans-for-the-future)
```
Comment (ricardosalveti) [inline release-notes/rn_v95.md:19]: FU/Fu

File: release-notes/rn_v95.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+  - [Known Issues](#known-issues)
+
+## Attention Points for Migration
+Things to be aware of when [updating LmP](https://docs.foundries.io/95/reference-manual/linux/linux-update.html) from the v94.y release:
+
+> - v95 is the first LmP release based on Yocto Project Scarthgap (5.0.8). So, when updating a FoundriesFactory from previous LmP versions the line `LAYERSERIES_COMPAT_meta-subscriber-overrides = "scarthgap"`  on `meta-subscriber-overrides/conf/layer.conf` will be required.
+> - When using the `lmp-tools/blob/master/scripts/update-factory-manifest` script to update from v94.y to v95 a merge conflict is expected due to mismatch on the `conf/keys` directory structure. The certificates files are expected to be present at `lmp-manifest/conf/keys` directory from the FoundriesFactory.
```
Comment (ricardosalveti) [inline release-notes/rn_v95.md:26]: So the expected merge path here is:
- Removal of all lmp-manifest/conf/keys after the merge (since that is done on new factories)
- Creation of the missing certificate files by hand and having them available in the lmp-manifest/factory-keys folder.

Comment (quaresmajose) [reply release-notes/rn_v95.md:26]: I would add: _Creation of the missing certificate files by hand or using the `lmp-tools`_
```bash
rm -rf lmp-manifest/conf/keys
cd lmp-manifest/factory-keys/uefi
git clone https://github.com/foundriesio/lmp-tools ../lmp-tools
../lmp-tools/security/uefi/gen_uefi_certs.sh
git restore . && git add . && git commit -s
```

File: release-notes/rn_v95.md [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+## Composectl Updates
+
+### New Features
+> - Added the publish command to package a Compose App as a container image and upload it to a container registry.
+> - Added the show command to print app manifest and Compose project details.
+> - Introduced App bundle indexing, which includes hash generation for each item in the bundle. The hashes are verified during app OTA updates, installation, and starting. Additionally, the maximum app bundle size is now limited to 1GB.
+
+### Bug Fixes
+> - Resolved minor issues, mainly related to app blob pruning and checking whether an app is installed and running.
```
Comment (kprosise) [inline release-notes/rn_v95.md:76]:
```suggestion
### Bug Fixes

> - Resolved minor issues, mainly related to app blob pruning and checking whether an app is installed and running.
```

File: release-notes/rn_v95.md [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+
+### Testing
+> - Added a Compose-based environment for development and end-to-end testing of composectl in a containerized setup.
```
Comment (kprosise) [inline release-notes/rn_v95.md:79]:
```suggestion
### Testing

> - Added a Compose-based environment for development and end-to-end testing of composectl in a containerized setup.
```

File: release-notes/rn_v95.md [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+>   - Linux-lmp updated to the v6.6.74 stable release
+>   - Linux-lmp-rt updated to the v6.6.65-rt47 stable release
+>   - Linux-lmp-ti-staging: updated to the 10.01.10 tag
+>   - TI BSP updated to the 11.00.01 release
+>   - NVIDIA Tegra BSP updated to the L4T R35.6.0 release
+>   - NXP BSP updated to the lf-6.6.52-2.2.0 release
+>   - u-boot-fio_imx: imx8mn and imx8mp now require CONFIG_SPL_MXC_OCOTP=y
+
+### Deprecation list
+> - BSP
```
Comment (kprosise) [inline release-notes/rn_v95.md:102]:
```suggestion
### Deprecation list

> - BSP
```

File: release-notes/rn_v95.md [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+>   - Xilinx BSP: Support was moved from `meta-lmp` to `meta-partner`
+>   - STM32 BSP: Support was moved from `meta-lmp` to `meta-partner`
+>   - iMX 8ULP: Support for this SoC was removed from LmP
+>   - iMX 7D/ULP: Support for this SoC was removed from LmP
+> - General
+>   - Jailhouse: Support for this package was removed from LmP (check the respective vendor BSP support)
+>   - Xenomai4: Support for this kernel feature was removed from LmP (check the respective vendor BSP support)
+>   - FIT_NODE_SEPARATOR: This variable is not used by LmP anymore
```
Comment (kprosise) [inline release-notes/rn_v95.md:110]:
```suggestion
>   - FIT_NODE_SEPARATOR: This variable is no longer used by LmP
```

File: release-notes/rn_v95.md [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+
+## Plans for the FUture
+>   - TI BSP: Support will be moved from `meta-lmp` to `meta-partner` in **v96**
```
Comment (kprosise) [inline release-notes/rn_v95.md:113]:
```suggestion
## Plans for the Future

>   - TI BSP: Support will be moved from `meta-lmp` to `meta-partner` in **v96**
```

File: release-notes/rn_v95.md (hunk 2 above)
Comment (angolini) [reply release-notes/rn_v95.md]: I don't like this huge number of lines for one item of release notes, and I agree they are important, but RN is way bigger than I would like for a quick doc.

What about we create:
[1] a blog post
[2] a FAQ entry

@quaresmajose @ricardosalveti @kprosise @vanmaegima

File: release-notes/rn_v95.md [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+  - [General Updates](#general-updates)
+    - [Deprecation list](#deprecation-list)
+  - [Plans for the FUture](#plans-for-the-future)
+  - [Known Issues](#known-issues)
+
+## Attention Points for Migration
+Things to be aware of when [updating LmP](https://docs.foundries.io/95/reference-manual/linux/linux-update.html) from the v94.y release:
+
+> - v95 is the first LmP release based on Yocto Project Scarthgap (5.0.8). So, when updating a FoundriesFactory from previous LmP versions the line `LAYERSERIES_COMPAT_meta-subscriber-overrides = "scarthgap"`  on `meta-subscriber-overrides/conf/layer.conf` will be required.
+> - When using the `lmp-tools/blob/master/scripts/update-factory-manifest` script to update from v94.y to v95, a merge conflict is expected due to a mismatch in the `conf/keys` directory structure. The certificates files are expected to be in the `lmp-manifest/conf/keys` directory from the FoundriesFactory.
```
Comment (vanmaegima) [inline release-notes/rn_v95.md:26]: does this applies to all previous versions or 94 only?
```suggestion
> - When using the `lmp-tools/blob/master/scripts/update-factory-manifest` script to update from LmP <= v94.y to v95, a merge conflict is expected due to a mismatch in the `conf/keys` directory structure. The certificates files are expected to be in the `lmp-manifest/conf/keys` directory from the FoundriesFactory.
```

File: release-notes/rn_v95.md [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+> - The default directory expected for the DTB files for Linux Kernel 6.6 has changed. Remove the `dir` from `KERNEL_DEVICETREE` items `<dir>/<dtb-name>`
+> - Aktualizr-Lite: Change in downgrade behavior, now downgrades are not allowed by default for the daemon
+> - u-boot-scr: Vendor prefix was removed from kernel-lmp-fitimage, so make sure to align the boot scripts to also remove the vendor prefix in the `fdtfile` variable.
+> - Jailhouse support in LmP is removed. In some rare cases, when a factory is created during a specific period of time, it might face the following error during the migration:
+> ```
+> Parsing recipes...ERROR: ParseError at /srv/oe/build/conf/../../layers/meta-subscriber-overrides/recipes-samples/images/lmp-factory-image.bb:30: Could not include required file recipes-samples/images/lmp-feature-jailhouse.inc
+> ERROR: Parsing halted due to errors, see error messages above
```
[the remark this answers is not in the document]
Comment (vanmaegima) [reply release-notes/rn_v95.md:33]: +1
if we don't want to cover the solution in RN, we should cover it in FAQ and link it here

File: release-notes/rn_v95.md (hunk 8 above)
Comment (quaresmajose) [reply release-notes/rn_v95.md:26]: > does this applies to all previous versions or 94 only?

all version, any update to v95 regardless of which version comes.

File: release-notes/rn_v95.md (hunk 2 above)
Comment (quaresmajose) [reply release-notes/rn_v95.md]: The customers custom keys are stored on `lmp-manifest/factory-keys/uefi` and the instructions keep everything that already exists here, adding only what is missing.
What we removed are the keys in the github public version of lmp `lmp-manifest/conf/keys` which should only be used for testing and not in private factories.

Comment (angolini) [reply release-notes/rn_v95.md]: I will have to handle this without specific command lines, because we cannot force a key rotation if they already have the keys only for the merge sake

And, we should cover more than only v94.y -> v95, because the list of removed keys changes depending on factory creation time

for example, my merge conflicts are related to the files:
	new file:   ../../conf/keys/uefi/noKEK.auth
	new file:   ../../conf/keys/uefi/noKEK.esl
	new file:   ../../conf/keys/uefi/noPK.auth
	new file:   ../../conf/keys/uefi/noPK.esl

which exists now, but when my factory was created.

the suggested command lines does not work for me, for example

File: release-notes/rn_v95.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (angolini) [reply release-notes/rn_v95.md:33]: I' m going to move this to a FAQ and include it in this PR to review

File: release-notes/rn_v95.md (hunk 8 above)
Comment (angolini) [reply release-notes/rn_v95.md:26]: all versions. I'm also changing this

File: release-notes/rn_v95.md (hunk 2 above)
Comment (quaresmajose) [reply release-notes/rn_v95.md]: > I will have to handle this without specific command lines, because we cannot force a key rotation if they already have the keys only for the merge sake

but the suggested command lines will not rotate any key, please note that it will only add new files in `lmp-manifest/factory-keys/uefi`.

let me place the full command list to update from a v94.1 factory created today to v95:

```bash
# get the lmp-tools to re-generate the conflicting keys
git clone https://github.com/foundriesio/lmp-tools /tmp/lmp-tools

cd lmp-manifest
git remote add up https://github.com/foundriesio/lmp-manifest.git
git fetch up
git merge up/main
# to see the conflicting keys
git status
git rm -f -r conf/keys
# re-generate the conflicting keys
cd factory-keys/uefi
/tmp/lmp-tools/security/uefi/gen_uefi_certs.sh
git status
# only keep the new files
git restore .
# add the new missing keys (none when updating from v94.1 to v95. but if the factory was created some time ago some will be added)
git add .
# finish the merge
git commit -s

# drop the lmp-tools
rm -rf /tmp/lmp-tools
```

But I agree with you that there will not be a command line that always works and that is why we cannot fix the update script.
However in my having some examples of how to fix it can help customers, but if you can describe it with just text even better.

Comment (vanmaegima) [reply release-notes/rn_v95.md]: Maybe an additional script in https://github.com/foundriesio/lmp-tools/tree/master/scripts to generate what's missing would be better than listing the steps (if possible)

File: release-notes/rn_v95.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (angolini) [reply release-notes/rn_v95.md]: this is covered by FAQ, please review

File: release-notes/rn_v95.md (hunk 2 above)
Comment (angolini) [reply release-notes/rn_v95.md]: 1- using gen_uefi_certs.sh will cause only uefi new files to be created, I decided that it would be better to "rotate" all the keys and see which new file is created
2- I tried the procedure (not the exact command lines) and the lmp-tools script still fails to merge, but the procedure works well for solving the merge conflict, so I think the next merge with the script will be fine (haven't tested)

Anyway, I suggested a FAQ, this is really tricky to document, so please review and provide insights

Comment (ricardosalveti) [reply release-notes/rn_v95.md]: Right, steps to fix the issue should indeed be in a FAQ, we should just say we have an issue here and point people to the location that describes the steps needed to fix it.

**Review by kprosise — APPROVED** [review]
lgtm, the troubleshooting addition was a good idea!

File: release-notes/rn_v95.md (hunk 2 above)
Comment (quaresmajose) [reply release-notes/rn_v95.md]: I think the FAQ approach covers our requirements.  I have also proposed a workaround for this type of failures https://github.com/foundriesio/lmp-tools/pull/52

**Review by quaresmajose — APPROVED** [review]
LGTM

**Review by vanmaegima — APPROVED** [review]
minor comment: missing entry to RN_v95 in the RN section
LGTM, though I could not check if the links to FAQ work

File: release-notes/rn_v95.md [hunk 10]
```diff
@@ -0,0 +1,157 @@
...
+### Testing
+
+> - Added a Compose-based environment for development and end-to-end testing of
+>  `composectl` in a containerized setup.
+> - Implemented a series of end-to-end tests, including tests for edge cases.
+
+## General Updates
+
+> - Yocto Project
+>   - LMP release based on the OE/Yocto 5.0.8 **Scarthgap** release
```
Comment (ricardosalveti) [inline release-notes/rn_v95.md:120]: We can probably set to 5.0.9 as we're expecting the wic backport to be done in oe-core.

**Review by ricardosalveti — APPROVED** [review]

**Comment by angolini** [comment]
> Perhaps it would be important to note that `ostree` has been updated to version 2024.5. The previous version was 2021.6 and this is a package with a significant impact on our distribution.

@quaresmajose , I understood that the addition of aklite and composectl would be enough to cover this suggestion. But now I'm unsure. Could you, please, review and provide feedback?

**Comment by quaresmajose** [comment]
> > Perhaps it would be important to note that `ostree` has been updated to version 2024.5. The previous version was 2021.6 and this is a package with a significant impact on our distribution.
> 
> @quaresmajose , I understood that the addition of aklite and composectl would be enough to cover this suggestion. But now I'm unsure. Could you, please, review and provide feedback?

I don't see mention of any of this but I also don't know how important this is for our customers. Better not mention nothing about the ostree.
